### DISCUSSION TITLE
Parameters for the root path?

### DISCUSSION THREAD

**Discussion by jaens** [TanStack/router#1076] [General]
Is it possible to add a common path prefix to **every** route with file-based routing?

eg. in a multi-tenant / multi-login app, every path starts with eg. `/user/12345/project/12345`.

As far as I see, this can currently be achieved by nesting every file under `/user/$userId/project/$projectId` or adding that as a prefix to every source file, both options have fairly... terrible ergonomics (eg. due to making unreadable imports in IDEs and command line paths).

**Comment by jaens** [comment]
Another related question would then be - how to specify `<Link>` components without having to repeat the initial prefix part everywhere. I guess every link would have to be relative? That, I think, prevents extracting common React components containing `<Link>`s since those could be instantiated at different levels in the path tree...

Every such link could theoretically be generated with a hook that uses eg. `useMatches()` to copy the first segments of the path, but can this approach support type safety?

**Comment by slhck** [comment]
Did you ever figure something out here? I am just getting started with this project and this seems like a dealbreaker for me if it can't be handled once, for all routes.

**Reply by jaens** [reply]
No. (although most popular routers don't handle this well either as far as I've looked)

Also encountered plenty of other problems (most of which have been solved by now), but still somewhat regretting the choice of Tanstack Router in a large-ish app, the TypeScript build times are getting insane.  
10 second **incremental** build just modifying a single query parameter for a link (on a high-end CPU, CI times are 5× worse).
